garage/tf/algos/ddpg.py

```diff
@@ -0,0 +1,371 @@
+
+                    observation = next_observation
+
+                    if terminal:
+                        episode_rewards.append(episode_reward)
+                        episode_steps.append(episode_step)
+                        episode_reward = 0.
+                        episode_step = 0
+                        episodes += 1
+
+                        observation = self.env.reset()
+                        if self.es:
+                            self.es.reset()
+
+                for train_itr in range(self.n_train_steps):
+                    if replay_buffer.size >= self.min_buffer_size:
+                        critic_loss, y, q, action_loss = self._learn()
+
+                        episode_actor_losses.append(action_loss)
+                        episode_critic_losses.append(critic_loss)
+                        epoch_ys.append(y)
+                        epoch_qs.append(q)
+
+            logger.log("Training finished")
+            if replay_buffer.size >= self.min_buffer_size:
+                logger.record_tabular('Epoch', epoch)
+                logger.record_tabular('Episodes', episodes)
+                logger.record_tabular('AverageReturn',
+                                      np.mean(episode_rewards))
+                logger.record_tabular('StdReturn', np.std(episode_rewards))
+                logger.record_tabular('AveragePolicyLoss',
+                                      np.mean(episode_actor_losses))
+                logger.record_tabular('AverageQFunctionLoss',
+                                      np.mean(episode_critic_losses))
+                logger.record_tabular('AverageQ', np.mean(epoch_qs))
+                logger.record_tabular('MaxQ', np.max(epoch_qs))
+                logger.record_tabular('AverageAbsQ', np.mean(np.abs(epoch_qs)))
+                logger.record_tabular('AverageY', np.mean(epoch_ys))
+                logger.record_tabular('MaxY', np.max(epoch_ys))
+                logger.record_tabular('AverageAbsY', np.mean(np.abs(epoch_ys)))
+
+                episode_rewards = []
+                episode_actor_losses = []
+                episode_critic_losses = []
+                epoch_ys = []
+                epoch_qs = []
+
+            logger.dump_tabular(with_prefix=False)
+            logger.pop_prefix()
+            if self.plot:
+                self.plotter.update_plot(self.actor, self.n_rollout_steps)
+                if self.pause_for_plot:
+                    input("Plotting evaluation run: Press Enter to "
+                          "continue...")
+
+        if self.plot:
+            self.plotter.shutdown()
+        if created_session:
+            sess.close()
+
+    def _initialize(self):
+        """Set up the actor, critic and target network."""
+        # Set up the actor and critic network
+        self.actor.build_net()
+        self.critic.build_net()
+
+        # Create target actor and critic network
+        target_actor = pickle.loads(pickle.dumps(self.actor))
```

Comment:
why `pickle.loads(pickle.dumps())`?

Reply:
To copy actor to target_actor. Same as `target_actor = copy(self.actor)`.

Reply:
If that is the case, then please use copy() so it is clearer.